plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -333,30 +329,13 @@ private function process_date_args(): void {
 			'date_paid'      => 'date_paid_gmt',
 			'date_completed' => 'date_completed_gmt',
 		);
-		$gmt_date_keys          = array_values( $local_to_gmt_date_keys );
-		$local_date_keys        = array_keys( $local_to_gmt_date_keys );
+
+		$gmt_date_keys   = array_values( $local_to_gmt_date_keys );
+		$local_date_keys = array_keys( $local_to_gmt_date_keys );
 
 		$valid_date_keys = array_merge( $gmt_date_keys, $local_date_keys );
 		$date_keys       = array_filter( $valid_date_keys, array( $this, 'arg_isset' ) );
 
-		// Process already passed date queries args.
-		if ( $this->arg_isset( 'date_query' ) && is_array( $this->args['date_query'] ) ) {
```

Comment:
note that block is refactored to a separate method `map_gmt_and_post_keys_to_hpos_keys`, only difference is, we now call this block on all passed array elements using array_walk, instead of just the outer one.